packages/ui-components/src/components/Message/Message.test.tsx

```diff
@@ -0,0 +1,162 @@
+    }
+  )
+
+  test("ensures the icon cannot shrink", () => {
+    render(<Message data-testid="my-message" />)
+    expect(screen.getByTestId("my-message")).toHaveClass("juno-message-info")
+    expect(screen.getByRole("img")).toHaveClass("jn-shrink-0")
+  })
+
+  // DISMISSIBLE BEHAVIOUR
+
+  test("renders a Message without a dismiss button by default", () => {
+    render(<Message data-testid="my-message" />)
+    expect(screen.queryByRole("button", { name: /close/i })).not.toBeInTheDocument()
+  })
+
+  test("renders a dismissible Message when 'dismissible' is true", async () => {
+    render(<Message data-testid="my-message" dismissible={true} />)
+
+    const user = userEvent.setup()
+    expect(screen.getByRole("button", { name: /close/i })).toBeInTheDocument()
+
+    // Ensure message is dismissed after clicking the button
+    await waitFor(async () => {
+      await user.click(screen.getByRole("button", { name: /close/i }))
+      expect(screen.queryByTestId("my-message")).not.toBeInTheDocument()
+    })
+  })
+
+  test("renders a Message without a dismiss button when 'dismissible' is false", () => {
+    render(<Message data-testid="my-message" dismissible={false} />)
+    expect(screen.queryByRole("button", { name: /close/i })).not.toBeInTheDocument()
+  })
+
+  test("calls 'onDismiss' handler when the Message is manually dismissed", async () => {
+    render(<Message data-testid="my-message" dismissible={true} onDismiss={mockOnDismiss} />)
+
+    const user = userEvent.setup()
+    expect(screen.getByRole("button", { name: /close/i })).toBeInTheDocument()
+    const dismissButton = screen.getByRole("button", { name: /close/i })
+
+    await waitFor(async () => {
+      await user.click(dismissButton)
+      expect(mockOnDismiss).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  // AUTO-DISMISS BEHAVIOUR
+
+  test("automatically dismisses the Message after the provided timeout", async () => {
+    render(<Message data-testid="my-message" autoDismiss={true} autoDismissTimeout={500} />)
+    expect(screen.getByTestId("my-message")).toBeInTheDocument()
+
+    // Wait to check that the message auto-dismissed
+    await new Promise((resolve) => setTimeout(resolve, 501))
+    expect(screen.queryByTestId("my-message")).not.toBeInTheDocument()
+  })
+
+  test("calls 'onDismiss' handler when the Message is automatically dismissed", async () => {
+    render(<Message data-testid="my-message" autoDismiss={true} autoDismissTimeout={500} onDismiss={mockOnDismiss} />)
+
+    await waitFor(() => {
+      expect(mockOnDismiss).toHaveBeenCalledTimes(1)
+    })
+  })
+
+  test("does not auto-dismiss when 'autoDismissTimeout' is set to zero", async () => {
+    render(<Message data-testid="my-message" autoDismiss={true} autoDismissTimeout={0} />)
+    expect(screen.getByTestId("my-message")).toBeInTheDocument()
+
+    // Wait to check that the message did not auto-dismiss
+    await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
Do we a promise here? is not waitFor or act are made for?

Reply:
Hmmm, I don't think so. Added due to this error (`waitFor` was there without `await`): 
<img width="717" alt="Screenshot 2024-10-14 at 14 00 57" src="https://github.com/user-attachments/assets/7b161967-0eda-407b-9839-ac17348c9013">
To my understanding, `waitFor` repeatedly calls what's inside until the condition is met or the timeout is reached. 
But it's still a `Promise`, so we need to stop execution in our code for it to be fulfilled.
But good point, now that I reflect on it... Can't we just do below? Much more readable.
```
await user.click(screen.getByRole("button", { name: /close/i }))
expect(screen.queryByTestId("my-message")).not.toBeInTheDocument()
```
However, the first way is used all over our code base...